File: src/java/queue/Queue.java
package queue;

import java.util.LinkedList;

public class Queue<T> {
    private LinkedList<T> elements;

    public Queue() {
        elements = new LinkedList<>();
    }

    // Add an element to the rear of the queue
    public void enqueue(T item) {
        elements.addLast(item);
    }

    // Remove and return the element at the front of the queue
    public T dequeue() {
        if (isEmpty()) {
            throw new IllegalStateException("Queue is empty");
        }
        return elements.removeFirst();
    }

    // Return the element at the front of the queue without removing it
    public T peek() {
        if (isEmpty()) {
            throw new IllegalStateException("Queue is empty");
        }
        return elements.getFirst();
    }

    // Check if the queue is empty
    public boolean isEmpty() {
        return elements.isEmpty();
    }

    // Get the size of the queue
    public int size() {
        return elements.size();
    }

    public static void main(String[] args) {
        Queue<Integer> queue = new Queue<>();

        queue.enqueue(1);
        queue.enqueue(2);
        queue.enqueue(3);

        System.out.println("Dequeued: " + queue.dequeue()); // Output: Dequeued: 1
        System.out.println("Peek: " + queue.peek()); // Output: Peek: 2
        System.out.println("Size: " + queue.size()); // Output: Size: 2
        System.out.println("Is empty? " + queue.isEmpty()); // Output: Is empty? false
    }
}

File: src/python/queue.py
class Queue:
    def __init__(self):
        self.elements = []

    def enqueue(self, item):
        self.elements.append(item)

    def dequeue(self):
        if self.is_empty():
            raise Exception("Queue is empty")
        return self.elements.pop(0)

    def peek(self):
        if self.is_empty():
            raise Exception("Queue is empty")
        return self.elements[0]

    def is_empty(self):
        return len(self.elements) == 0

    def size(self):
        return len(self.elements)
    def main():
        queue = Queue()

        # Enqueue elements
        queue.enqueue(10)
        queue.enqueue(20)
        queue.enqueue(30)

        # Dequeue an element
        dequeued = queue.dequeue()
        print("Dequeued element:", dequeued)  # Output: Dequeued element: 10

        # Peek at the front element
        front_element = queue.peek()
        print("Front element:", front_element)  # Output: Front element: 20

        # Check if the queue is empty
        empty = queue.is_empty()
        print("Is the queue empty?", empty)  # Output: Is the queue empty? False

        # Get the size of the queue
        size = queue.size()
        print("Size of the queue:", size)  # Output: Size of the queue: 2


    if __name__ == "__main__":
        main()
